### PR TITLE
nx-X11's Xserver: Drop outdated Xserver extensions.

### PR DESCRIPTION
**Pull request by sunweaver**
The X.Org project has dropped various of their Xserver extensions since nx-X11 got forked from X.org.

WIth this commit, we drop these outdated extensions from nx-X11. As libnx-ext6 is only used by nxagent and nowhere else, we are safe with dropping the Xlib counterparts of those extensions, as well (where they exist).

Among the dropped extensions are:
- Double Buffer extension (DBE/Xdbe)
- TOG-CUP (colormap) protocol extension (Xcup)
- X Extended Visual Information extension (XEvi)
- X11 Double-Buffering, Multi-Buffering, and Stereo extension (Xmbuf)
- XTrap extension

Also, we drop various XF86 video mode extensions plus the MIT-Misc extension.

Before reviewing this PR, the following PR needs to be reviewed:
https://github.com/ArcticaProject/nx-libs/pull/93
https://github.com/ArcticaProject/nx-libs/pull/102

### CODE REVIEW

**Comment by sunweaver** [comment]
Before reviewing this PR, the following PR needs to be reviewed:
https://github.com/ArcticaProject/nx-libs/pull/102

File: debian/rules [hunk 1]
```diff
@@ -45,19 +45,11 @@ override_dh_install:
...
-	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xf86vmstr.h
-	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xf86misc.h
-	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xf86dga1.h
-	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xf86dgastr.h
-	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xf86dga.h
-	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xf86dga1str.h
-	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xf86mscstr.h
 	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/bigreqstr.h
 	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/xcmiscstr.h
+	rm -R debian/tmp/usr/include/$(DEB_BUILD_MULTIARCH)/nx-X11/extensions/Xdbeproto.h
```
Comment (Ionic) [inline debian/rules]: Is this correct? Do we still need Xdbeproto, if the extension has been removed?

Ah, wait, Xdbeproto is part of libXext and we're still using the internal libXext version, not the system-wide library and headers, right?

File: nx-X11/config/cf/README [hunk 2]
```diff
@@ -172,15 +172,10 @@ X11.tmpl provides defaults for the following variables:
 	BDFTOSNFFILT		command to run bdftosnf
 	BaseExtensionDefines	universal extensions to use
 	BinDir			directory in which to install programs
-	BuildAppgroup		build AppGroup extension in server
-	BuildCup		build Colormap Utilization extension in server
 	BuildDBE		build DOUBLE-BUFFER extension
```
Comment (Ionic) [inline nx-X11/config/cf/README:175]: Is BuildDBE still necessary?

Probably yes, it's used to add targets from `nx-X11/programs/Xserver/dbe/*`, which is really the MI-DBE component.

File: nx-X11/config/cf/X11.tmpl [hunk 3]
```diff
@@ -251,18 +227,9 @@ XORGRELSTRING = XorgManVersionString
 #ifndef BuildScreenSaverExt
 #define BuildScreenSaverExt	NO
 #endif
-#ifndef BuildXF86VidModeExt
-#define BuildXF86VidModeExt	NO
-#endif
-#ifndef BuildXF86MiscExt
-#define BuildXF86MiscExt	NO
-#endif
```
Comment (Ionic) [inline nx-X11/config/cf/X11.tmpl:259]: BuildXF86MiscExt is being removed here, but it is still referenced in `nx-X11/include/extensions/Imakefile`.

Generally, xf86misc isn't deprecated as far as I know, but still part of libXext (or split out of it.) Do you want to remove it now in order to replace libXext later with the system library?

**Comment by Ionic** [comment]
In `nx-X11/programs/Xserver/Xext/Imakefile`: you removed `$(PNRXOBJS)` from `SOBJS`, this is most likely wrong (given it was kept in all other places.)

Sorry for not being able to make an in-line comment, github won't let me.

**Comment by Ionic** [comment]
In `nx-X11/programs/Xserver/Xext/panoramiX.c`: stray whitespace at EOL inserted.

**Comment by Ionic** [comment]
In `nx-X11/programs/Xserver/Xext/panoramiXprocs.c:` a lot of changes to REGION_... macros have been reverted, most likely also a mistake?

**Comment by Ionic** [comment]
`XFreeXDGA`, `XF86VIDMODE`, `XF86MISC` undefined in `nx-X11/programs/Xserver/hw/nxagent/Imakefile` and `nx-X11/programs/Xserver/mi/Imakefile`. We can probably drop that too, now that these macros aren't used anymore.

from sunweaver: This PR does not touch nx-X11/programs/Xserver/hw/nxagent/ at all, so postponing cleanup here.

**Comment by Ionic** [comment]
`MITMiscDefines`, `BuildMITMiscExt` could probably also be removed from `nx-X11/config/cf/X11.tmpl`.

**Comment by Ionic** [comment]
Stray left over `$(FONTCACHEINCLUDES)` in `nx-X11/programs/Xserver/Xext/extmod/Imakefile`?

**Comment by Ionic** [comment]
`LBX` macro check left in `nx-X11/include/extensions/security.h`.

**Comment by Ionic** [comment]
Regarding `XAPPGROUP`, `XEVIE`, `LBX`: these are still checked for in the modified NX files such as `nx-X11/programs/Xserver/hw/nxagent/NXdispatch.c` and others. Can't this be dropped as well? Given we're not defining the macros anywhere, these code blocks are essentially dead as well?

from sunweaver: This PR does not touch nx-X11/programs/Xserver/hw/nxagent/ at all, so postponing cleanup here.

**Comment by Ionic** [comment]
I think that's been my review for now, I have no further comments to add.

File: debian/rules (hunk 1 above)
Comment (sunweaver) [inline debian/rules]: You are actually right, we have to keep this (for Xserver/dbe/*) and it actually belongs into (Debian pkg) nx-x11proto-xext-dev and (RPM pkg) nx-x11-proto. Have a fix for that locally now.

File: nx-X11/config/cf/README (hunk 2 above)
Comment (sunweaver) [inline nx-X11/config/cf/README:175]: Exactly. Still necessary.

File: nx-X11/config/cf/X11.tmpl (hunk 3 above)
Comment (sunweaver) [inline nx-X11/config/cf/X11.tmpl:259]: Good catch. No, afaict xf86misc has been removed from xorg-server.

Fixed locally.

**Comment by sunweaver** [comment]
@Ionic: I have force-pushed my changes as proposed by you to the PR branch. Please go ahead and merge. Thanks.

**Comment by Ionic** [comment]
> XFreeXDGA, XF86VIDMODE, XF86MISC undefined in nx-X11/programs/Xserver/hw/nxagent/Imakefile and nx-X11/programs/Xserver/mi/Imakefile. We can probably drop that too, now that these macros aren't used anymore.
> 
> from sunweaver: This PR does not touch nx-X11/programs/Xserver/hw/nxagent/ at all, so postponing cleanup here.

Yep, but they are also being undefined via -U in `nx-X11/programs/Xserver/mi/Imakefile`. I think that can be dropped.

After rebasing this, I'll merge the PR. :)

**Comment by sunweaver** [comment]
@Ionic: ok. After having pushed some bad rebased state of the PR branch, this has been fixed now. Patchset applies cleanly and contains only commits belonging to this PR.

Please merge.

**Pull request merged**